### ISSUE TITLE
Meteor improvements

### ISSUE THREAD

**Issue by Technici4n** [AppliedEnergistics/Applied-Energistics-2#5875]
Opening this issue to discuss potential improvements to meteors. We will not remove them, but we are open to feedback regarding possible improvements.

A few ideas to get the discussion started: compass points to nearest chest instead of pointing to nearest sky stone block, more ae2 loot in meteors, per-player loot for servers, ...?

**Comment by mindforger** [comment]
Is it in any way possible to detect if an area of blocks has never been touched since generation? You could spawn it as kind of retrogen when somebody holds the compass in it's hands but underground as option for servers

**Comment by Technici4n** [comment]
So, some kind of retrogen? I don't think it's realistically possible, but wouldn't that only help for people installing the mod after some playtime anyway?

**Comment by shartte** [comment]
We converted meteorites into structures in 1.15 or 1.16 (I don't quite remember). There might actually be a general mod for retro-genning structures that were added later, I haven't looked into it, but the API for it certainly is generic enough to do it.

My idea for the compass was to embrace the little "hack" we did for adding fluix to the meteorite (we should btw. get a cobblestony-fluix-texture for that, I think) and make the compass track fluix (which in the "lore" is high-energy) instead of the sky-stone. That would make it far easier to remove/mine the fluix and have the compass point to the next meteorite.

Alternatives: 
Make it a special fluix encased goodie block that drops its contents upon being broken, instead of a chest and make that the target of the compass. That would make it trivial to implement too.

**Comment by mindforger** [comment]
my idea was to improve the situation where people join an already active server and can not find any new meteors because others already heavily explored the world

**Comment by Chad90b** [comment]
> Opening this issue to discuss potential improvements to meteors. We will not remove them, but we are open to feedback regarding possible improvements.

This is a shame. I'm now running on 1.18.1 and there are meteors abso-f-everywhere. I would rather cheat in the required resources at this point because every other biome has a huge ugly hole in it. in 2018 we were promised a setting to adjust spawn rate but I do not see this option in the v10 alpha-7 config files.

Is there a way to reduce the meteor spawn rate?

I would also love to see smaller craters in general. I don't know why they need to be several chunks wide.. Isn't it a glorified loot chest?

**Comment by shartte** [comment]
>  Is there a way to reduce the meteor spawn rate?

Currently not. It's something that could be investigated however. You can disable them in their entirety or for specific biomes though.

> I would also love to see smaller craters in general. I don't know why they need to be several chunks wide.. Isn't it a glorified loot chest?

It's a glorified loot chest that is a signature feature of the mod I guess ;-)

**Comment by Ommina** [comment]
Is it also possible to enable / disable meteors on a ~~Dimension~~ ~~World~~ Level basis?  I was always a fan of spawning them in a convenient mining dimension (see above), while leaving the Overworld untouched.  (Presumably if the mining dimension uses its own biome, one could blacklist everything BUT that particular biome, but that's a bit unwieldy.)

**Comment by Tenchu98** [comment]
It would be nice to have a way to force the meteors to spawn in a specific dim instead of the overworld. As an exmplae I am currently making a 1.16.5 pack that I want AE to be late game, and so would prefer to make the meteors spawn in either the end or one of the divine rpg dimensions.

**Comment by Tenchu98** [comment]
I would also suggest that a new tag be added to skystone, since it obviously survived rentry and impact, it should be a witherproof block as well.

**Comment by Ommina** [comment]
While I get the logic, that makes it a pretty early-game witherproof block for those using default meteor spawning.

**Comment by Tenchu98** [comment]
to be fair wither proof blocks tend to be somewhat mid game in most packs, and the same mining level as skystone.
EIO only needs dark steel and obsidian to make a witherproof block.

**Comment by Ommina** [comment]
Are three diamonds a stick especially mid-game though?

I dunno.  I guess it very much comes down to whatever the individual pack maker is trying to achieve.  Balance doesn't exist in a vacuum.  (Which is why https://github.com/AppliedEnergistics/Applied-Energistics-2/pull/4837 disappoints me, but I think I'm taking this a bit too far off topic, so I'll leave it at that.)

**Comment by Tenchu98** [comment]
perhaps just having options to be able to enable such features

**Comment by ndizazzo** [comment]
> That would make it far easier to remove/mine the fluix and have the compass point to the next meteorite.

I spend so much time mining the entire meteorite only to learn there's _some_ skystone block somewhere under the terrain that I've missed and can't find, this would be a very welcome change

**Comment by Technici4n** [comment]
Going to close this for now as we will implement some of this in 1.19. If you have a specific feature request for 1.19 (only once we start releasing builds publicly!), please open a dedicated issue.
